### ISSUE TITLE
Ops relevant for Supporting Dropout

### ISSUE THREAD

**Issue by cyang49** [torch-spyre/torch-spyre#117]
### What feature would you like to be added?

Dropout is implemented as something like [this](https://d2l.ai/chapter_multilayer-perceptrons/dropout.html)
```
def dropout_layer(X, dropout):
    if dropout == 1: return torch.zeros_like(X)
    mask = (torch.rand(X.shape) > dropout).float()
    return mask * X / (1.0 - dropout)
```
This requires some currently unsupported ops, I know that constant operand is planned, but unsure about others. Starting this thread for discussion and tracking
- [ ] constant operand  #238 
- [ ] on device randomized tensor creation
- [x]  `gt` (greater than) and other comparisons
- [x] `torch.where` support
- [ ] boolean reinterpretation as floating point

### Why is this feature needed?

A common layer in models

**Comment by cyang49** [comment]
eval mode dropout is supported through torch.nn.functional.dropout
